### DISCUSSION TITLE
Comp efficiency - loopy belief propagation speed comparison for marginals?

### DISCUSSION THREAD

**Discussion by mercicle** [pgmpy/pgmpy#1437] [Q&A] [answered]
Hey all,

Wondering if you've done a loopy belief propagation speed comparison for marginals between pgmpy and:

1. https://github.com/mbforbes/py-factorgraph/
2. https://pomegranate.readthedocs.io/en/latest/FactorGraph.html

Right now using py-factorgraph, do you have a more efficient implementation? @ankurankan would love your thoughts thx!

**Comment by ankurankan** [comment, marked as answer]
@mercicle I think pomegranate might be the computationally fastest option as it has cython implementation. pgmpy focuses more on extensibility at the cost of being computationally more expensive.
